src/backend/core/grafana_events.py

```
@@ -0,0 +1,191 @@
+    try:
+        with _guardrail_lock:
+            attempts = _guardrail_attempts.get(guardrail, 0) + 1
+            if passed:
+                _guardrail_attempts[guardrail] = 0
+            else:
+                _guardrail_attempts[guardrail] = attempts
+        if passed:
+            emit_event("guardrail_passed", guardrail=guardrail, attempts=attempts)
+        else:
+            emit_event("guardrail_retry", guardrail=guardrail, attempt=attempts)
+    except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)